caravel/assets/visualizations/filter_box.jsx

```diff
@@ -10,22 +10,80 @@ import '../stylesheets/react-select/select.less';
 
 import './filter_box.css';
 
+const TIME_CHOICES = [
+  '1 hour ago',
+  '12 hours ago',
+  '1 day ago',
+  '7 days ago',
+  '28 days ago',
+  '90 days ago',
+  '1 year ago',
+];
+
+const propTypes = {
+  origSelectedValues: React.PropTypes.object,
+  filtersChoices: React.PropTypes.object,
+  onChange: React.PropTypes.func,
+  showDateFilter: React.PropTypes.bool,
+};
+
+const defaultProps = {
+  origSelectedValues: {},
+  onChange: () => {},
+  showDateFilter: false,
+};
+
 class FilterBox extends React.Component {
```

Comment:
i think this component should be in it's own file in the components directory, and imported for use in this visualization file.